### ISSUE TITLE
Wrong line endings when using getall -f csv

### ISSUE THREAD

**Issue by rubendura** [fugue/credstash#102]
TL;DR When using getall -f csv the line endings are always set to \r\n, which is not appropriate depending on the environment. The issue seems to be related to the csv.writer used in the csv_dump() function in crestash. There is more information about the appropriate way to create those wirters in the docs: https://docs.python.org/3/library/csv.html#csv.writer. There is a newline parameter which seems to be sort of required, but using the value specified in there ('') doesn't seem to work either. I got it to work on my distro by setting newline to None. I still don't fully understand why.

Context and further details:

I've been trying to debug a really weird issue I found while trying to load all secrets at once.

I am launching my application using a script that loads the credstash secrets one by one, sets those on variables (with hardcoded names in the script) and exports those. Then it execs my application.

As I keep adding secrets startup times started to lag a bit, a few seconds per new variable. I found out that the issue was the python interpreter startup times: loading it once or twice it's no problem, but loading it 10 or 12 times (once per key) it was definitely noticeable.

I tried a new strategy: using getall and some scripting. Using a while loop to which I pipe my secrets I export those variables. For simplicity the credstash key names are the same as the names or the variables being exported. Although I struggled a bit to get it to work (I hate shell scripting) in the end I managed to get it to work with the following snippet (environment: alpine linux 3.4 using busybox's dash, sh, shell):

``` bash
PARAMS="-r $AWS_DEFAULT_REGION -t $CREDSTASH_TABLE"
credstash $PARAMS getall -f csv > secrets.txt

while IFS=, read -r varname varvalue
do

    if [ -z $varname ]
    then
        continue
    fi

    export $varname=$varvalue

done < secrets.txt

exec "$@"
```

(Further info: I tried not having to output the results of credstash using a here doc instead of redirecting stdin from a file, but I've already lost many hours trying to figure out why the 💩 it doesn't work. Some help would be appreciated!)

The problem is that now all my variable values end in '\r'. Weird. I could trim it using tr or something like that, but I don't get why that is when Python now technically uses Universal Line Endings.

**Comment by gene1wood** [comment]
I can confirm this bug. This looks [like a good approach to a fix](https://stackoverflow.com/a/29976091/168874)

**Comment by artburkart** [comment]
I added the line change that @gene1wood suggested. If anything else is required to get this merged, please let me know! :)

**Comment by gene1wood** [comment]
@artburkart Thanks for PR #179 !
